core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/FcCidEncoder.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.core.service.jsonrpc;
+
+import tech.pegasys.web3signer.core.util.Blake2b;
+import tech.pegasys.web3signer.core.util.ByteUtils;
+
+import java.math.BigInteger;
+
+import org.apache.tuweni.bytes.Bytes;
+
+public class FcCidEncoder {
+
+  private static final BigInteger FC_HASHING_ALGO_CODE = BigInteger.valueOf(0xb220);
```

Comment:
a comment that these are hard-coded magic numbers from lotus?

Reply:
Replacing all these constants and using the Java CID library. That should make all of this clearer.